README.md

```diff
@@ -1,50 +1,134 @@
 [![Build Status](https://img.shields.io/endpoint?url=https%3A%2F%2Fci.ocamllabs.io%2Fbadge%2Frealworldocaml%2Fmdx%2Fmain&logo=ocaml)](https://ci.ocamllabs.io/github/realworldocaml/mdx)
 
-## mdx -- executable code blocks inside markdown files
+## mdx -- executable code blocks inside documentation
 
-`mdx` allows to execute code blocks inside markdown files.
-There are (currently) two sub-commands, corresponding
-to two modes of operations: pre-processing (`ocaml-mdx pp`)
-and tests (`ocaml-mdx test`).
+`mdx` allows to execute code blocks inside markdown and mli documentation
+to help keeping them up to date.
 
-The pre-processor mode allows to mix documentation and code,
-and to practice "literate programming" using markdown and OCaml.
+Use the
+[dune stanza](https://dune.readthedocs.io/en/latest/dune-files.html#mdx-since-2-4)
+to enable it on your documentation.
 
-The test mode allows to ensure that shell scripts and OCaml fragments
-in the documentation always stays up-to-date.
-
-`mdx` is released as a single binary (called `ocaml-mdx`) and
-can be installed using opam:
+`mdx` is released on opam and can be installed by running:
 
 ```sh
 $ opam install mdx
 ```
 
-If you want to contribute or hack on the project, please see the
+If you want to contribute to the project, please see the
 [CONTRIBUTING.md](CONTRIBUTING.md).
 
+### Basic Usage
+
+You can use MDX with your Markdown or `.mli` documentation, which ensures
+code in multi-line or verbatim code blocks is correct.
+
+To enable MDX, you need to add `(mdx)` stanzas to the right `dune` files. Before
+that you must enable MDX for your project by adding the following to
+your `dune-project`:
+
+```
+(using mdx 0.2)
+```
+
+Then add the following in the relevant `dune` file:
+```
+(mdx)
+```
+That enables MDX on all markdown files in the folder.
+The MDX stanza can be further configured. Please visit the relevant section of
+[dune's manual](https://dune.readthedocs.io/en/latest/dune-files.html#mdx-since-2-4)
+for more information.
```

Comment:
We should probably note that this needs dune 2.4 explicitly. And mdx 0.2 even needs dune 3.0.